voluncheer/opportunityboard/forms/postanopportunity.py

```diff
@@ -26,30 +27,77 @@ class PostAnOpportunityForm(forms.ModelForm):
 
     class Meta:
         model = Opportunity
+
         fields = (
-            "category",
+            "organization",
             "title",
+            "category",
+            "subcategory",
+            "subsubcategory",
             "description",
+            "staffing",
             "date",
-            "duration",
+            "end",
             "address_1",
             "address_2",
             "is_published",
             "photo",
         )
 
+        labels = {"subcategory": "Subcategory 1", "subsubcategory": "Subcategory 2"}
+
+        widgets = {"organization": forms.HiddenInput()}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["subcategory"].queryset = Subcategory.objects.none()
+        self.fields["subsubcategory"].queryset = Subsubcategory.objects.none()
+
+        if "category" in self.data:
```

Comment:
Are you trying to conduct test here? What exactly are you trying to accomplish?

Reply:
We need to override `__init__` for the form to render properly. For example, when a form for a new opportunity loads, the subcategories drop-down should have no options because an opportunity hasn't been selected yet. We then populate the drop down with a list of subcategories based on the category selection.

That particular test is if we would reload to the same form page after submitting it, which we are not doing currently. For us, after submitting the form, we redirect to the org. profile page. So theoretically we could probably delete everything up to the `elif`, but it's also not hurting us and covers us in case we decide to change down the road. Good catch, I'm open to either option.

See this tutorial for reference: https://simpleisbetterthancomplex.com/tutorial/2018/01/29/how-to-implement-dependent-or-chained-dropdown-list-with-django.html

Reply:
Let's keep it like this for right now since it's working.